cron/data/blob.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2021 Security Scorecard Authors
+//
+// Licensed under the Apache License, Vershandlern 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permisshandlerns and
+// limitathandlerns under the License.
+
+package data
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ossf/scorecard/cron/config"
+	"gocloud.dev/blob"
+)
+
+const (
+	filePrefixFormat = "2006.01.02/150405/"
+)
+
+var (
+	errorWrite      = errors.New("blobWriter.Write")
+	errorWriteClose = errors.New("blobWriter.Close")
+	errorOpenBucket = errors.New("blob.OpenBucket")
+)
+
+func WriteToBlobStore(ctx context.Context, bucketURL, filename string, data []byte) error {
+	bucket, err := blob.OpenBucket(ctx, bucketURL)
+	if err != nil {
+		return fmt.Errorf("%w: %v", errorOpenBucket, err)
+	}
+	defer bucket.Close()
+
+	blobWriter, err := bucket.NewWriter(ctx, filename, nil)
+	if err != nil {
+		return fmt.Errorf("%w", err)
+	}
+	if _, err = blobWriter.Write(data); err != nil {
+		return fmt.Errorf("%w: %v", errorWrite, err)
+	}
+	if err := blobWriter.Close(); err != nil {
+		return fmt.Errorf("%w: %v", errorWriteClose, err)
+	}
+	return nil
+}
+
+func GetBlobFilename(filename string, datetime time.Time) string {
+	return datetime.Format(filePrefixFormat) + filename
+}
+
+func GetShardNumFilename(datetime time.Time) string {
```

Comment:
Docs on this function would be helpful. Probably how and the data is going to be sharded.

Reply:
I want to avoid obvious comments in code. The fn does precisely as the name suggests, i.e - given a `time.Time` value, returns a filename `string` which in this case, is a pre-defined metadata file called `.shard_num`. It does not impact data sharding.